recipes/fast-dds/all/conanfile.py

```diff
@@ -116,28 +112,47 @@ def package(self):
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "fastdds")
 
-        # component fastrtps
-        self.cpp_info.components["fastrtps"].set_property("cmake_target_name", "fastrtps")
-        self.cpp_info.components["fastrtps"].set_property("cmake_target_aliases", ["fastdds::fastrtps"])
-        self.cpp_info.components["fastrtps"].libs = collect_libs(self)
-        self.cpp_info.components["fastrtps"].requires = [
-            "fast-cdr::fast-cdr",
-            "asio::asio",
-            "tinyxml2::tinyxml2",
-            "foonathan-memory::foonathan-memory",
-        ]
-        if self.settings.os in ["Linux", "FreeBSD", "Neutrino"]:
-            self.cpp_info.components["fastrtps"].system_libs.append("pthread")
-        if self.settings.os in ["Linux", "FreeBSD"]:
-            self.cpp_info.components["fastrtps"].system_libs.extend(["rt", "dl", "atomic", "m"])
-        elif self.settings.os == "Windows":
-            self.cpp_info.components["fastrtps"].system_libs.extend(["iphlpapi", "shlwapi", "mswsock", "ws2_32"])
-            if self.options.shared:
-                self.cpp_info.components["fastrtps"].defines.append("FASTRTPS_DYN_LINK")
-        if self.options.with_ssl:
-            self.cpp_info.components["fastrtps"].requires.append("openssl::openssl")
+        if Version(self.version) < "3.0.0":
+            # component fastrtps
+            self.cpp_info.components["fastrtps"].set_property("cmake_target_name", "fastrtps")
+            self.cpp_info.components["fastrtps"].set_property("cmake_target_aliases", ["fastdds::fastrtps"])
+            self.cpp_info.components["fastrtps"].libs = collect_libs(self)
```

Comment:
@perseoGI I'd like to avoid using `collect_libs`. I think that we could apply the same as below? (code not tested 🙏 )

```suggestion
            if self.settings.compiler == "msvc":
                version = Version(self.version)
                self.cpp_info.components["fastrtps"].libs = [f"fastrtps{'d' if self.settings.build_type == 'Debug' else ''}-{version.major}.{version.minor}"]
            else:
                self.cpp_info.components["fastrtps"]libs = ["fastrtps"]
```

Reply:
I know, I'm aware of it and I dont like it either. 
The idea with this PR is adding the new major and leaving intact the previous major, avoiding any side effects.